backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/select/BpkSelect.kt

```diff
@@ -0,0 +1,44 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.select
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import net.skyscanner.backpack.compose.select.internal.BpkSelectImpl
+
+sealed interface BpkSelectState {
+
+    object Default : BpkSelectState
+
+    object Disabled : BpkSelectState
+
+    data class Error(val message: String) : BpkSelectState
+}
+
+@Composable
+fun BpkSelect(
```

Comment:
we also need the other version which is the "self serve" one where we don't need to handle the option selection

Reply:
Sorry, I have not got it. Could you explain it more?

Reply:
there's two API tables in Figma - we need to implement both. One is this one and the other one is for just having a click listener and the actual selection etc is done by the consumer

Reply:
will add another version without options.